### DISCUSSION TITLE
ERROR: Call to undefined function osc_csrfguard_generate_token()

### DISCUSSION THREAD

**Discussion by eurobank** [mindstellar/shopclass#404] [Q&A] [answered]
I DESPERATELY need some help on this for a plugin and Osclass 5.1.

This 

2021-10-01 05:31:59 - ERROR: Call to undefined function osc_csrfguard_generate_token() 

at

`<a class="btn"onclick="return delete_dialog('<?php echo $id; ?>');" href="<?php list($csrfname, $csrftoken) = osc_csrfguard_generate_token(); echo osc_admin_base_url(true).'?page=items&spam=delete&item='.$id.'&CSRFName='.$csrfname.'&CSRFToken='.$csrftoken; ?>"><?php _e('Delete Ad', 'spamprotection'); ?></a>`

apparently `function osc_csrfguard_generate_token` has been replaced. 

How can i fix this?

**Comment by navjottomer** [comment, marked as answer]
Something like this.
```php
<a class="btn" onclick="return delete_dialog('<?php echo $id; ?>');" href="<?php  echo osc_admin_base_url(true).'?page=items&spam=delete&item='.$id.'&'. osc_csrf_token_url(); ?>"><?php _e('Delete Ad', 'spamprotection'); ?></a>
```
